source/common/http/hash_policy.cc

```diff
@@ -33,21 +36,33 @@ class HeaderHashMethod : public HashMethodImplBase {
     }
   }
 
+  const std::string HEADER_VALUE_SEPARATOR = ",";
+
   absl::optional<uint64_t> evaluate(const Network::Address::Instance*,
                                     const RequestHeaderMap& headers,
                                     const HashPolicy::AddCookieCallback,
                                     const StreamInfo::FilterStateSharedPtr) const override {
     absl::optional<uint64_t> hash;
 
-    // TODO(mattklein123): Potentially hash on all headers.
     const auto header = headers.get(header_name_);
     if (!header.empty()) {
-      if (regex_rewrite_ != nullptr) {
-        hash = HashUtil::xxHash64(regex_rewrite_->replaceAll(header[0]->value().getStringView(),
-                                                             regex_rewrite_substitution_));
-      } else {
-        hash = HashUtil::xxHash64(header[0]->value().getStringView());
+      std::vector<std::string> header_values;
+      header_values.reserve(header.size());
+      for (size_t i = 0; i < header.size(); i++) {
+        std::string value(header[i]->value().getStringView());
```

Comment:
It doesn't look like you actually need the separator or the string copy of value here.

Reply:
Emm... I didn't get this, I still need the separator for the case said in the next line.

Reply:
I think the logic might be a little borked here. If we assume for simplicity that regex_write_ is nullptr, and you have value "foo", we'll be pushing "foo\n" into the array. How does that help the hasher to have that separator? Nothing is coming after the separator.